lib/facter/consul_version.rb

```diff
@@ -0,0 +1,15 @@
+# consul_version.rb
+
+Facter.add(:consul_version) do
+  setcode do
+    if Facter.value(:kernel) == 'Linux'
+      version = Facter::Util::Resolution.exec('consul --version')
+      version = version.lines.first.split[1].tr('v','')
+    elsif Facter.value(:kernel) == 'windows'
+      version = Facter::Util::Resolution.exec('"C:\\Program Files\\Consul\\consul.exe" --version')
```

Comment:
If there is no easy way to parameterize it then yea might want to remove it.  My company is working on the Windows support for this module and are baking it in our preprod environments currently so hopefully we'll have that merged in soon.